src/statsig.erl

```diff
@@ -1,140 +1,48 @@
 -module(statsig).
 
--behaviour(gen_server).
+-behaviour(application).
 
 -export(
   [
-    init/1,
-    terminate/2,
-    handle_call/3,
-    handle_cast/2,
-    handle_info/2,
-    check_gate/3,
+    start_link/0,
+    start/2,
+    stop/1,
+    check_gate/2,
+    log_event/3,
     log_event/4,
-    log_event/5,
-    flush/1
+    flush/0
   ]
 ).
 
--import(utils, [get_timestamp/0]).
--import(network, [start/0, stop/0, request/4]).
--import(evaluator, [eval_gate/3]).
--import(logging, [get_event/4, get_exposure/4]).
+start_link() ->
+  gen_server:start_link(?MODULE, [], []).
 
-init([{apiKey, ApiKey}]) ->
-  network:start(),
-  Body = network:request(ApiKey, "download_config_specs", #{}, true),
-  if
-    Body == false ->
-      network:stop(),
-      {stop, "Initialize Failed"};
+start(_Type, _Params) ->
+  {ok, ApiKey} = application:get_env(statsig, statsig_api_key),
+  {ok, Pid} = gen_server:start(statsig_server, [{api_key, ApiKey}], []),
+  register(statsigsdk, Pid).
 
-    true ->
-      {
-        ok,
-        [
-          {configSpecs, jiffy:decode(Body, [return_maps])},
-          {logEvents, []},
-          {apiKey, ApiKey}
-        ]
-      }
-  end.
+-spec check_gate(map(), binary()) -> boolean().
+check_gate(User, Gate) -> 
+  Pid = whereis(statsigsdk),
+  gen_server:call(Pid, {User, Gate}).
```

Comment:
Similarly, if the process is registered, it's enough to call `gen_server:call(statsigsdk, {User, Gate})`.